modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/snapshot/CurrencySnapshotAcceptanceManager.scala

```diff
@@ -264,6 +352,94 @@ object CurrencySnapshotAcceptanceManager {
     ): SortedSet[SharedArtifact] =
       sharedArtifactsForAcceptance
 
+    private def acceptTokenLocks(
+      incomingTokenLocks: SortedSet[Signed[TokenLock]],
+      lastSyncGlobalSnapshotEpochProgress: EpochProgress
+    ): SortedSet[Signed[TokenLock]] =
+      if (incomingTokenLocks.isEmpty) {
+        incomingTokenLocks
+      } else {
+        incomingTokenLocks.filter(itl => itl.unlockEpoch >= lastSyncGlobalSnapshotEpochProgress)
+      }
+
+    private def acceptTokenUnlocks(
+      incomingTokenUnlocks: SortedSet[TokenUnlock],
+      activeTokenLocksReferences: List[TokenLockReference]
+    ): SortedSet[TokenUnlock] =
+      if (incomingTokenUnlocks.isEmpty) {
+        incomingTokenUnlocks
+      } else {
+        incomingTokenUnlocks.filter { itu =>
+          activeTokenLocksReferences.contains(itu.lockReference)
+        }
+      }
+
+    private def updateActiveTokenLocks(
+      maybeActiveTokenLocks: Option[SortedMap[Address, SortedSet[Signed[TokenLock]]]],
+      acceptedTokenLocks: SortedSet[Signed[TokenLock]],
+      acceptedTokenUnlocks: SortedSet[TokenUnlock]
+    )(implicit hasher: Hasher[F]): F[Option[SortedMap[Address, SortedSet[Signed[TokenLock]]]]] =
+      if (acceptedTokenLocks.isEmpty && acceptedTokenUnlocks.isEmpty) {
+        maybeActiveTokenLocks.pure
+      } else {
+        val incomingTokenLocks = acceptedTokenLocks.groupBy(_.source).toSortedMap
+        val activeTokenLocks = maybeActiveTokenLocks.getOrElse(SortedMap.empty[Address, SortedSet[Signed[TokenLock]]])
+
+        val allTokenLocks = (incomingTokenLocks.keySet ++ activeTokenLocks.keySet).toList.map { address =>
+          address -> (incomingTokenLocks.getOrElse(address, SortedSet.empty[Signed[TokenLock]]) ++ activeTokenLocks.getOrElse(
+            address,
+            SortedSet.empty[Signed[TokenLock]]
+          ))
+        }.toSortedMap
+
+        if (acceptedTokenUnlocks.isEmpty) {
+          allTokenLocks.some.pure
+        } else {
+          allTokenLocks.toList.traverse {
+            case (address, set) =>
+              set.toList.filterA { signedTokenLock =>
+                TokenLockReference.of(signedTokenLock).map { tokenLockReference =>
+                  !acceptedTokenUnlocks.exists(_.lockReference == tokenLockReference)
```

Comment:
instead of negating, you can use `acceptedTokenUnlocks.forall(_.lockReference =!= tokenLockReference)`.
please please remember about the Eq comparison (`===` or `=!=`)